plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/generated/CircleManager.kt

```diff
@@ -229,22 +226,22 @@ class CircleManager(
       }
     }
 
-  /**
-   * Create the source for managed annotations
-   *
-   * @return the GeoJsonSource created
-   */
-  override fun createSource(): GeoJsonSource {
-    return geoJsonSource(sourceId) { data("") }
-  }
-
   /**
    * Create the layer for managed annotations
    *
    * @return the layer created
    */
   override fun createLayer(): CircleLayer {
-    return CircleLayer(layerId, sourceId)
+    return circleLayer(layerId, sourceId) {
+      annotationConfig?.geoJsonOptions?.let { geoJsonOptions ->
+        geoJsonOptions.minZoom?.let {
+          minZoom(it)
+        }
+        geoJsonOptions.maxZoom?.let {
+          maxZoom(it)
+        }
+      }
+    }
```

Comment:
Both source as layer have the concept of minZoom and maxZoom (they do different things though) but conceptually, the minZoom/maxZoom provided to the geoJsonOptions should only be applied to the source. I do think we can ticket a feature request for exposing this additionally to layer but as separate properties.